agent/src/util/discovery_operator.rs

```diff
@@ -689,11 +709,21 @@ pub mod start_discovery {
                 sleep_duration = Duration::from_millis(100);
             }
 
-            if tokio::time::timeout(sleep_duration, stop_discovery_receiver.recv())
-                .await
-                .is_ok()
+            if let Ok(result) =
+                tokio::time::timeout(sleep_duration, stop_discovery_receiver.recv()).await
             {
-                trace!("do_discover_on_discovery_handler - received message to stop discovery for {} Discovery Handler at endpoint {:?}", dh_details.name, dh_details.endpoint);
+                if let Ok(Some(config_id)) = result {
```

Comment:
same request for a little comment describing how send(None) and send(configid) are handled here